### PR TITLE
Remove OpenSSL, Cleanup & Increase Coverage

### PR DESCRIPTION
**Pull request by ikeydoherty**
Much the same as the last Pull Request, but pulling in a synced, updated Nica, and also merging the unit tests in, as this is the most active consumer now of Nica. i.e. its developed backwards, we fix it here, and sent it out into the Nica repo.

Thus, tests should be here now. Local coverage is now 69% but I'm sure coveralls will say it reduced to 11.2% or something equally crazy. Best to check this one commit-by-commit, the overall file change looks larger than it really is.

### CODE REVIEW

**Comment by ikeydoherty** [comment]
@bryteise psst - it's over here xD

**Comment by bryteise** [comment]
Indeed, I didn't expect a new patchset so quickly =).

**Comment by bryteise** [comment]
The libnica change surprised me actually.

**Comment by ikeydoherty** [comment]
Yeah I mean it's up to you whether you want the new nica stuff in now, or to hold it off until after a tag is done

**Comment by bryteise** [comment]
Hrm thinking about it. Darn travis making it a pain if you don't vendor C deps that aren't in Ubuntu =P.

**Comment by ikeydoherty** [comment]
Lol yea. So what I'd do then is merge everything *before* update_format.sh change and leave the tail end for another day. The main thing is those CBM-specific cleans and nuking of OpenSSL for a footprint reduction

**Comment by ikeydoherty** [comment]
Or.. well. Basically the other PR..

**Comment by bryteise** [comment]
Heh well I'll see how it goes. Long day for me ahead I can tell =).

**Comment by ikeydoherty** [comment]
Sowwy :(

File: src/bootman/files.c [hunk 1]
```diff
@@ -105,19 +105,27 @@ char *get_sha1sum(const char *p)
...
-        h1 = get_sha1sum(p1);
-        if (!h1) {
+        if (!cbm_mapped_file_open(p1, m1)) {
                 return false;
         }
-        h2 = get_sha1sum(p2);
-        if (!h2) {
+
+        if (!cbm_mapped_file_open(p2, m2)) {
+                return false;
```
Comment (bryteise) [inline src/bootman/files.c]: I love autofree

File: src/bootloaders/syslinux.c [hunk 2]
```diff
@@ -196,7 +194,7 @@ static bool syslinux_set_default_kernel(const BootManager *manager, const Kernel
                 kname_copy = NULL;
                 free(boot_options);
                 boot_options = NULL;
-                if (streq(k->path, kernel->path)) {
```
Comment (bryteise) [inline src/bootloaders/syslinux.c]: Ah nice improvement.

File: src/bootloaders/syslinux.c [hunk 3]
```diff
@@ -172,19 +172,17 @@ static bool syslinux_set_default_kernel(const BootManager *manager, const Kernel
                                 abort();
                         }
                 }
-                if (streq(k->path, kernel->path)) {
```
Comment (bryteise) [inline src/bootloaders/syslinux.c]: Test for kernel not needed, kernel is checked outside the for loop and isn't changed.

File: src/bootloaders/systemd-class.c [hunk 4]
```diff
@@ -95,7 +95,8 @@ bool sd_class_init(const BootManager *manager, BootLoaderConfig *config)
         prefix = boot_manager_get_prefix((BootManager *)manager);
 
         /* ia32 paths */
-        if (!asprintf(&ia32_source, "%s/%s/%s", prefix, sd_config->efi_dir, sd_config->ia32_blob)) {
+        if (asprintf(&ia32_source, "%s/%s/%s", prefix, sd_config->efi_dir, sd_config->ia32_blob) <
+            0) {
```
Comment (bryteise) [inline src/bootloaders/systemd-class.c]: Oh wow that column limit is painfully bad...

**Comment by bryteise** [comment]
Okay reviewed, only a day late =P. Only minor nits I think so nearly ready for release.

File: src/bootloaders/systemd-class.c (hunk 4 above)
Comment (ikeydoherty) [inline src/bootloaders/systemd-class.c]: Yeeerp.. Thinking of 120-150.. unsure.

File: src/bootloaders/syslinux.c (hunk 3 above)
Comment (ikeydoherty) [inline src/bootloaders/syslinux.c]: Ack

File: tests/check-files.c [hunk 5]
```diff
@@ -70,7 +70,6 @@ static Suite *core_suite(void)
 
         s = suite_create("bootman_files");
         tc = tcase_create("bootman_files");
-        tcase_add_test(tc, bootman_hash_test);
```
Comment (ikeydoherty) [inline tests/check-files.c]: Yeah I cocked that one up and will put it earlier in the next PR

File: src/bootman/files.c (hunk 1 above)
Comment (ikeydoherty) [inline src/bootman/files.c]: Lol same :D Considering promoting `CbmMappedFile` directly into Nica tbh

**Comment by ikeydoherty** [comment]
Cheers, I'll do a fresh round and pull out the nitpicks. It was on the fat side hence wanting extra eyes on it. Shall I drop the nica changes from round 2?

**Comment by bryteise** [comment]
Nah turned out I liked the nica changes lol. We'll rip it out eventually but for now is better to just update I think.

File: src/bootloaders/syslinux.c (hunk 3 above)
Comment (ikeydoherty) [inline src/bootloaders/syslinux.c]: Hm have to disagree, we still continue even without a default kernel, which may be NULL, and try to use it within that loop (regardless of its NULL or not).

Comment (bryteise) [inline src/bootloaders/syslinux.c]: Ah totally, I misread the check.

**Pull request closed without merging**